internal/top/top.go

```diff
@@ -28,55 +31,59 @@ type TopResponseError struct {
 	Message string `json:"Message"`
 }
 
-func Process() ([]PrettyResponse, error) {
+func Process(topChanReq chan<- map[string]TopData, errChan chan<- error) {
 	client := getClient()
 	tResponse := TopResponse{}
-	PrettyResp := []PrettyResponse{}
+	topData := make(map[string]TopData)
 	apiConf := conf.ApiConfig[conf.TopApi]
 	pageNumInt, err := strconv.Atoi(apiConf.Options[conf.PageParam])
 	if err != nil {
-		return nil, err
+		errChan <- err
 	}
+	var wg sync.WaitGroup
+	wg.Add(pageNumInt)
 	for i := 0; i < pageNumInt; i++ {
 		currentPage := strconv.Itoa(i)
 		req, err := tResponse.CreateRequest(apiConf, currentPage)
 		if err != nil {
-			return nil, err
+			errChan <- err
 		}
 		response, err := tResponse.GetResponse(req, client)
 		if err != nil {
-			return nil, err
+			errChan <- err
 		}
 		err = json.Unmarshal(response, &tResponse)
 		if err != nil {
-			return nil, err
+			errChan <- err
 		}
 		err = tResponse.ValidateResponse()
 		if err != nil {
-			return nil, err
+			errChan <- err
 		}
 		for _, v := range tResponse.Data {
-			PrettyResp = append(PrettyResp, PrettyResponse{Symbol: v.CoinInfo.Symbol, Rank: len(PrettyResp) + 1})
+			topData[v.CoinInfo.Symbol] = TopData{Symbol: v.CoinInfo.Symbol, Rank: len(topData) + 1}
 		}
+		wg.Done()
```

Comment:
if you got an error and you skip/stop executing,  wg.Done() never release wg.Add(), mean you get a deadlock on wg.Wait()